fvm/src/machine/engine.rs

```diff
@@ -182,13 +186,19 @@ impl Engine {
         let dummy_memory = Memory::new(&mut dummy_store, MemoryType::new(0, Some(0)))
             .expect("failed to create dummy memory");
 
+        let actor_redirect = ec
+            .actor_redirect
+            .as_ref()
+            .map(|v| v.iter().cloned().collect());
```

Comment:
If we're not overriding anything, we might as well just have an empty hashmap.

```suggestion
            .as_ref()
            .unwrap_or_default().iter().cloned().collect();
```

Reply:
sure, do you want to drop the option from the data structure?
I thought it would be faster to `None` check instead of a get in an empty map, but if there is no difference we might as well.

Reply:
An option prevents us from hashing on access when there are no redirects set.